nixos/modules/services/security/sshguard.nix

```diff
@@ -146,6 +148,7 @@ in {
           ] ++ (map (name: "-w ${escapeShellArg name}") cfg.whitelist));
         in "${pkgs.sshguard}/bin/sshguard ${args}";
         Restart = "always";
+        X-ReloadIfChanged = configFile;
```

Comment:
This isn't valid:

https://github.com/NixOS/nixpkgs/blob/638936f65785aaebfb96990604b984d0eb0bfedd/nixos/modules/system/activation/switch-to-configuration.pl#L221-L224

You're supposed to use `systemd.services.<name>.reloadIfChanged` instead of directly using `X-ReloadIfChanged`.

... but do you want this? Do you understand the implications of `reloadIfChanged`?

Reply:
only just started in the nix world, so i'm probably missing a lot. pretty sure we'd need some kind of reload/restart when the config has changed, though the way fail2ban does it seems a lot better than this here. would it be acceptable to just do the same in sshguard?

Reply:
@symphorien it looks like there is a bug in the `nagios` module that @pennae has found:

https://github.com/NixOS/nixpkgs/blob/6e4f8b06f5e396df212474f497ba77d5157a6ef2/nixos/modules/services/monitoring/nagios.nix#L204

Are you able to fix that up when you get a chance?